src/main/scala/org/apache/spark/sql/pulsar/PulsarMetadataReader.scala

```diff
@@ -453,6 +454,7 @@ private[pulsar] case class PulsarMetadataReader(
             .topic(tp)
             .startMessageId(MessageId.earliest)
             .startMessageIdInclusive()
+            .loadConf(readerConfig)
```

Comment:
the `loadConf` method and other set methods should be mutually exclusive.

User either provides a `ReaderConfig` to load or set detailed configs with these setter method

Reply:
Can we provide a priority to the user, the configuration based on the `pulsar.reader.` prefix will override the corresponding reader configuration